sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/ECDsaKeyVault.cs

```diff
@@ -0,0 +1,165 @@
+        }
+
+        /// <inheritdoc />
+        protected override byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm)
+        {
+            ValidateKeyDigestCombination(KeySize, hashAlgorithm);
+
+            using (IncrementalHash hash = IncrementalHash.CreateHash(hashAlgorithm))
+            {
+                hash.AppendData(data, offset, count);
+                return hash.GetHashAndReset();
+            }
+        }
+
+        /// <inheritdoc/>
+        public override bool VerifyHash(byte[] hash, byte[] signature)
+        {
+            CheckDisposed();
+            ValidateKeyDigestCombination(KeySize, hash.Length);
+
+            return _publicKey.VerifyHash(hash, signature);
+        }
+
+        ///<inheritdoc/>
+        public override ECParameters ExportParameters(bool includePrivateParameters)
+        {
+            if (includePrivateParameters)
+                throw new CryptographicException("Private keys cannot be exported by this provider");
+
+            return _publicKey.ExportParameters(false);
+        }
+
+        ///<inheritdoc/>
+        public override ECParameters ExportExplicitParameters(bool includePrivateParameters)
+        {
+            if (includePrivateParameters)
+                throw new CryptographicException("Private keys cannot be exported by this provider");
+
+            return _publicKey.ExportExplicitParameters(false);
+        }
+
+        /// <summary>
+        /// Importing parameters is not supported.
+        /// </summary>
+        public override void ImportParameters(ECParameters parameters) =>
+            throw new NotSupportedException();
+
+        /// <summary>
+        /// Key generation is not supported.
+        /// </summary>
+        public override void GenerateKey(ECCurve curve) =>
+            throw new NotSupportedException();
+
+        /// <inheritdoc/>
+        public override string ToXmlString(bool includePrivateParameters)
+        {
+            if (includePrivateParameters)
+                throw new CryptographicException("Private keys cannot be exported by this provider");
+
+            return _publicKey.ToXmlString(false);
+        }
+
+        /// <summary>
+        /// Importing parameters from XML is not supported.
+        /// </summary>
+        public override void FromXmlString(string xmlString) =>
+            throw new NotSupportedException();
+
+        private static void ValidateKeyDigestCombination(int keySizeBits, int digestSizeBytes)
+        {
+            if (keySizeBits == 256 && digestSizeBytes == 32 ||
+                keySizeBits == 384 && digestSizeBytes == 48 ||
+                keySizeBits == 521 && digestSizeBytes == 64)
+            {
+                return;
+            }
+
+            throw new NotSupportedException($"The key size '{keySizeBits}' is not valid for digest of size '{digestSizeBytes}' bytes.");
```

Comment:
Thinking about this more and looking at exceptions on `RSACryptoServiceProvider`, I wonder if it would make more sense - outside of `ArgumentException` and derivatives - to wrap other exceptions including `RequestFailedException` - as inner exceptions in a `CryptographicException`. There's not a lot more thrown from those, apart from `NotSupportedException` that only seems to be thrown when a platform doesn't support some algorithm for some operation. @schaabs what do you think? It would provide a more expected API, though I know in the past we've often discussed not really caring about consistent exceptions because it's rare to want to catch specific exception types. Of course, @clairernovotny, you'd also have a lot of insight into that as well being on the .NET team, who we've tried to (mostly) mimic. What are your thoughts?

Reply:
That makes sense to me; that within the context of an RSA/ECDsa call, that any failures would be wrapped in a `CryptographicException`.